src/basic/device-nodes.h

```diff
@@ -22,5 +22,13 @@
 #include <stddef.h>
 #include <sys/types.h>
 
+#include "macro.h"
+#include "stdio-util.h"
+
 int encode_devnode_name(const char *str, char *str_enc, size_t len);
 int whitelisted_char_for_devnode(char c, const char *additional);
+
+#define SYS_BLOCK_PATH_MAX(suffix)                                      \
+        (strlen("/sys/dev/block/") + DECIMAL_STR_MAX(dev_t) + 1 + DECIMAL_STR_MAX(dev_t) + strlen(suffix))
+#define xsprintf_sys_block_path(buf, suffix, devno)                     \
+        xsprintf((buf), "/sys/dev/block/%u:%u" suffix, major(devno), minor(devno))
```

Comment:
maybe `… %u:%u%s", major(devno), minor(devno), strempty(suffix))`? (to permit NULL suffix, as suggested above)

Reply:
OK.